### ISSUE TITLE
[nREPL] Direct connection failed

### ISSUE THREAD

**Issue by abhigogna** [clojure-emacs/cider#1885]
Cider errors out and throws Direct connection failed.
Its been working fine but recently started throwing this error. 
I downloaded Emacs 25.1.1 from emacs for mac (https://emacsformacosx.com/)

## Expected behavior
Run cider

## Actual behavior
This is from emacs-toggle-debugger
Debugger entered--Lisp error: (error "[nREPL] Direct connection failed")
  signal(error ("[nREPL] Direct connection failed"))
  error("[nREPL] Direct connection failed")
  nrepl--direct-connect("localhost" 51710)
  nrepl-connect(nil 51710)
  nrepl-start-client-process(nil 51710 #<process nrepl-server>)
  nrepl-server-filter(#<process nrepl-server> "nREPL server started on port 51710 on host 127.0.0.1 - nrepl://127.0.0.1:51710\n")

## Steps to reproduce the problem
Just cider-jack-in. I have to restart the computer to make it work again.

## Environment & Version information
This is what env returned
TERM_PROGRAM=iTerm.app
TERM=xterm-color
SHELL=/bin/bash
CLICOLOR=1
TMPDIR=/var/folders/f6/0n9q4r0d1hs0jqym3whfbxf1rr2n6w/T/
Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.jM5Lw0kwYb/Render
TERM_PROGRAM_VERSION=3.0.2
TERM_SESSION_ID=w0t0p1:FDADADD0-11A8-46E7-8A21-FE11D36C4FAE
http_proxy=http://company-proxy:8080
USER=user
BOTO_CONFIG=~/.boto/boto.cfg

LSCOLORS=ExFxBxDxCxegedabagacad
PATH=/usr/local/sbin:/usr/local/Cellar/rabbitmq/3.5.6/sbin:/usr/local/bin:/Users/gognaab/.cabal/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/FINRA/CocoaDialog.app/Contents/MacOS:/usr/local/git/bin:/opt/oracle/instantclient_11_2:/opt/subversion/bin:/Library/TeX/texbin
PWD=/Users/gognaab/my-emacs/emacs

JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_66.jdk/Contents/Home
LANG=en_US.UTF-8
ITERM_PROFILE=Default
XPC_FLAGS=0x0
PS1=\[\e[0;33m\]\u\[\e[0m\]:\[\e[0;34m\]\W\[\e[0m\]\$
PS2=>>
XPC_SERVICE_NAME=0
SHLVL=1

COLORFGBG=7;0
no_proxy=169.254.169.254, .finra.org, localhost, 127.0.0.1

ITERM_SESSION_ID=w0t0p1:FDADADD0-11A8-46E7-8A21-FE11D36C4FAE
LOGNAME=gognaab
GROOVY_HOME=/usr/local/opt/groovy/libexec
DISPLAY=/private/tmp/com.apple.launchd.BouBMxJuQ9/org.macosforge.xquartz:0

### CIDER version information

*cider 0.14.0
CIDER's REPL is launched. Here's an example:*

```
;; Starting nREPL server via /usr/local/bin/lein update-in :dependencies conj \[org.clojure/tools.nrepl\ \"0.2.12\"\ \:exclusions\ \[org.clojure/clojure\]\] -- update-in :plugins conj \[cider/cider-nrepl\ \"0.14.0\"\] -- repl :headless...
```

### Emacs version
Version 25.1.1 (x86_64-apple-darwin13.4.0, NS appkit-1265.21 Version 10.9.5 (Build 13F1911)) of 2016-09-20

### Operating system

OS X 10.10.5 "Yosemite".

**Comment by dpsutton** [comment]
@abhigogna I've got a couple questions for you if you can help me understand.

1) Is this a boot or lein project? 
2) Can you run the project from the command line without CIDER, ie, lein run or boot <task> or whatever
3) When you say you have to restart to make it work, does that means that you get this error *sometimes* but not always, and when you get the error, you can restart the machine and you expect it to work afterwards?
4) Is it particular projects that cause this error or does it happen across all projects?

**Comment by abhigogna** [comment]
I think I have found the problem. I have a tunneling software, Appgate on my computer, and it was running while I was trying to connect. Sometimes appgates stop working and its child nodes need to be killed. It was running while I was trying to connect to the nRepl through cider. After I killed it, everything started working again.

So, I guess this is not a cider/emacs issue. Thanks for you quick response.
